docker/docker_tools.py

```diff
@@ -248,12 +276,23 @@ def build(
             print(f"Unable to run git: {e} Installing...")
             xqt("sudo apt-get install -y --no-install-recommends git")
 
+        # Check if Clone-All Flag is set
```

Comment:
Also check that the other clone flags aren't set. If they are, issue a warning that their values are ignored, or even quit with an error.

Reply:
I'm confused because default repositories is "RunestoneInteractive" and if the ```-c / --clone-all``` flag is even touched then it will set all the  repositories to attempt cloning with specified username. Are you saying that I should implement a check for each repository that will error out/exit if it cannot clone from specified user's remote repositories?

Reply:
Here, I'm asking for a CLI consistency check: it doesn't make sense (to me) to specify both `--clone-all=my_repo` while also specifying `--clone_rc=my_other_repo`. I can see three solutions:

1. In the help text, state that value supplied in `--clone-rs`, `--clone-bs`, and `--clone-rc` will override the value supplied in `--clone-all`. Also issue a warning when this happens.
2. Issue an error, telling user that the same value was specified twice.

I'll leave it to you to decide which is best.

On your second point: yes, you definitely want to error if a git clone from the specified repo fails.

Reply:
> Here, I'm asking for a CLI consistency check: it doesn't make sense (to me) to specify both --clone-all=my_repo while also specifying --clone_rc=my_other_repo. I can see three solutions:

Okay, I think I see your confusion. You are only meant to specify ```--clone-all``` if you wish to clone all the repositories with your remote repositories. Hypothetically, if you only want to clone BookServer from your own remote repository then you would specify ```--clone-bs```. 

I think I will be more specific within the ```--help``` stating the outcome of using ```--clone-all``` in tandem with any other clone flag. I could also provide some examples within ```--help``` if that would help with increasing the thought behind its functionality.
<br><br/>
> On your second point: yes, you definitely want to error if a git clone from the specified repo fails.

Addressed this with ```sys.exit()``` on another comment.

Reply:
Addressed this with commit 9e1c34ae7e9361b70d46ca267b4d471bbc08e98d